src/ol/View.js

```diff
@@ -1377,7 +1402,18 @@ class View extends BaseObject {
       }
     }
 
-    this.fitInternal(geometry, options);
+    return new Promise((resolve) => {
+      const performFitInternal = () => {
+        this.fitInternal(geometry, options);
+        resolve();
```

Comment:
Instead of resolving here, the promise should be resolved when the fit animation is complete. See `options.callback`.

Reply:
I was thinking that the promise returned by fit() is resolved when fitting can happen and that the animation callback is a different matter. If the promise should resolve only after the (possible) animation completes, then I think we should remove or at least deprecate the animation callback. What do you think?

Reply:
How is this going? We are interested in this feature as well. Thanks!